plugin/src/main/java/com/playmonumenta/epicstructures/managers/RespawningStructure.java

```diff
@@ -398,11 +399,25 @@ public void setSpawnerBreakTrigger(SpawnerBreakTrigger trigger) {
 	}
 
 	public boolean registerZone() {
-		ZoneLayer<RespawningStructure> zoneLayer = mPlugin.mRespawnManager.mZoneLayer;
-		return zoneLayer.addZone(mOuterBounds.mLowerCorner.clone(),
-		                         mOuterBounds.mUpperCorner.clone(),
-		                         mName,
-		                         new LinkedHashSet<String>(),
-								 this);
+		ZoneLayer zoneLayerInside = mPlugin.mRespawnManager.mZoneLayerInside;
+		ZoneLayer zoneLayerNearby = mPlugin.mRespawnManager.mZoneLayerNearby;
+		Map<Zone, RespawningStructure> structuresByZone = mPlugin.mRespawnManager.mStructuresByZone;
+
+		Zone insideZone = new Zone(zoneLayerInside,
+		                           mInnerBounds.mLowerCorner.clone(),
+		                           mInnerBounds.mUpperCorner.clone(),
+		                           mName,
+		                           new LinkedHashSet<String>());
+		Zone nearbyZone = new Zone(zoneLayerNearby,
+		                           mOuterBounds.mLowerCorner.clone(),
+		                           mOuterBounds.mUpperCorner.clone(),
+		                           mName,
+		                           new LinkedHashSet<String>());
+		zoneLayerInside.addZone(insideZone);
+		zoneLayerNearby.addZone(nearbyZone);
+		structuresByZone.put(insideZone, this);
+		structuresByZone.put(nearbyZone, this);
+
```

Comment:
Can we move these to a setter method in the respawn manager? Weird to access the hashmap directly outside of that class, though it does work.